### PR TITLE
Don't save timingwindows if some are invalid

### PR DESCRIPTION
**Pull request by cquiroz**
image (1)

### CODE REVIEW

**Comment by toddburnside** [comment]
Because this uses `DatePicker`, I guess we can't solve the problem with validators like we did for, for example, hour angles:
image

But, could we prevent it from happening by adding min/max dates to the DatePickers properties? In other words, prevent the error state from ever happening?

**Comment by cquiroz** [comment]
Not sure if we can do that, maybe we can if there is a max or min date property?

**Comment by cquiroz** [comment]
https://reactdatepicker.com/#example-min-date

**Comment by cquiroz** [comment]
Great suggestion, we can use min/max dates and a bit of css

image

File: explore/src/main/scala/explore/common/TimingWindowsQueries.scala [hunk 1]
```diff
@@ -14,29 +14,42 @@ import explore.syntax.ui.*
...
+                 WHERE = obsIds.toList.toWhereObservation.assign,
+                 SET = ObservationPropertiesInput(
+                   timingWindows = value.filter(_.isValid).map(_.toInput).assign
+                 )
+               )
+             )
+             .toastErrors
+             .void
+         else
+           ToastCtx[F].clear() *> ToastCtx[F]
```
Comment (toddburnside) [inline explore/src/main/scala/explore/common/TimingWindowsQueries.scala:50]: Can we ever hit this now?

**Review by toddburnside — APPROVED** [review]
Nice. LGTM.

File: explore/src/main/scala/explore/common/TimingWindowsQueries.scala (hunk 1 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/common/TimingWindowsQueries.scala:50]: I'd rather leave the error handling on just in case

**Pull request merged**